astropy/coordinates/builtin_frames/astrometric.py

```diff
@@ -124,9 +123,9 @@ def astrometric_to_astrometric(from_astrometric_coord, to_astrometric_frame):
         # Otherwise, the transform occurs just by setting the new origin.
         return to_astrometric_frame.realize_frame(from_astrometric_coord.cartesian)
```

Comment:
I just started to write this test, then realized it is much more difficult in this version of the API. In the old version of the API, you could make a new frame ``AstrometricICRS(origin=None)``, in the new API, it is less clear what the base frame for ``AstrometricFrame(origin=None)`` should be. Maybe we should remove support for this sort of transform.